### PR TITLE
Add Virmid (target-lib, pipeline, and tool)

### PR DESCRIPTION
**Pull request by smondet**
[Review on Reviewable](https://reviewable.io/reviews/hammerlab/biokepi/5)

### CODE REVIEW

File: src/lib/biokepi_pipeline.ml [hunk 1]
```diff
@@ -202,6 +207,11 @@ let rec to_json: type a. a t -> json =
           "Configuration", Strelka.Configuration.to_json config;
           "input", to_json bam_pair;
         ]]
+    | Virmid (config, bam_pair) ->
```
Comment (arahuja) [inline src/lib/biokepi_pipeline.ml]: OOC can you explain what's going on in these lines with the `[`Assoc [`?

File: src/lib/biokepi_target_library.ml [hunk 2]
```diff
@@ -829,6 +829,79 @@ The usage is:
...
+    let open Configuration in 
+    let name = Filename.basename result_prefix in
+    let result_file suffix = result_prefix ^ suffix in
+    let output_file = result_file "-somatic.vcf" in
+    let output_prefix = "virmid-output" in
+    let work_dir = result_file "-workdir" in
+    let reference_fasta =
+      Machine.get_reference_genome run_with `B37 |> Reference_genome.fasta in
+    let virmid_tool = Machine.get_tool run_with "virmid" in
+    let virmi_somatic_broken_vcf =
```
Comment (arahuja) [inline src/lib/biokepi_target_library.ml]: virmi[d]_somatic_broken_vcf

File: src/lib/biokepi_pipeline.ml (hunk 1 above)
Comment (smondet) [inline src/lib/biokepi_pipeline.ml]: The idea is to build values of the type: `Yojson.Basic.json`: http://mjambon.com/yojson-doc/Yojson.Basic.html

File: src/lib/biokepi_target_library.ml [hunk 3]
```diff
@@ -829,6 +829,79 @@ The usage is:
...
+          && sh (String.concat ~sep:" " ([
+              "java -jar $VIRMID_JAR -f";
+              "-w"; work_dir;
+              "-R"; reference_fasta#product#path;
+              "-D"; tumor#product#path;
+              "-N"; normal#product#path;
+              "-t"; Int.to_string processors;
+              "-o"; output_prefix;
+            ] @ Configuration.render configuration))
+          && shf "sed 's/\\(##INFO.*Number=A,Type=String,\\) /\\1/' %s > %s"
```
Comment (arahuja) [inline src/lib/biokepi_target_library.ml]: comment?

File: src/lib/biokepi_pipeline.ml [hunk 4]
```diff
@@ -272,3 +282,8 @@ let compile_variant_caller_step ~work_dir ~machine (t: vcf t) =
     let tumor = compile_aligner_step ~work_dir ~is:`Tumor ~machine tumor_t in
     Strelka.run
       ~run_with:machine ~normal ~tumor ~result_prefix ~processors:4 ~configuration ()
+  | Virmid (configuration, Bam_pair (normal_t, tumor_t)) ->
+    let normal = compile_aligner_step ~work_dir ~is:`Normal ~machine normal_t in
+    let tumor = compile_aligner_step ~work_dir ~is:`Tumor ~machine tumor_t in
+    Virmid.run
+      ~run_with:machine ~normal ~tumor ~result_prefix ~processors:4 ~configuration ()
```
Comment (arahuja) [inline src/lib/biokepi_pipeline.ml]: is `processors:4` passed to virmid at some point?

File: src/lib/biokepi_target_library.ml (hunk 3 above)
Comment (smondet) [inline src/lib/biokepi_target_library.ml]: The comment is up there, line 878, this call tries to fix the VCF (which still seems to break PyVCF's parsing for some other reason, I decided to wait until Cycledash figure where's the problem).

File: src/lib/biokepi_pipeline.ml (hunk 4 above)
Comment (arahuja) [inline src/lib/biokepi_pipeline.ml]: Nevermind, I see it now `"-t"; Int.to_string processors;`

**Pull request merged**